### PR TITLE
feat: teams avatars

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at 1f9d07a</samp>

This pull request adds a new field `userTeams` to the `Team` type in the GraphQL schema and updates the related files and components to handle the new field. The purpose is to enable the team management feature in the `journeys-admin` app, which allows users to view, join, and invite other users to teams. The pull request also improves the test coverage and functionality of the `team` module in the `api-journeys` service, and enhances the `Avatar` component to accept custom styles.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/33034379/todos/6363227500)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] it should display members of the team on teams select per the members on the team
- [x] it should display members of the selected active team on team menu bar on the top right

# Walkthrough

### <samp>🤖 Generated by Copilot at 1f9d07a</samp>

*  Add `userTeams` field to `Team` type in GraphQL schema and update related queries, resolvers, and models ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R1672), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R1991), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1211), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-41d403336e29ab179760e88aa306f9e877b82482443bb2d0f29a8125cac766acR6), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-8b8ecb917dba7942b0198ce80f346fc7f4b2640448416ae151bddbac82584e08L15-R33), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-9ec6366e6bee0170a2999e3923081140c4ca4d0d4a73cdfc7366f2ef00071640R40-R48))
* Implement `userTeams` field resolver for `Team` type and update `team` module resolvers and tests ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-e81088563f09768bf87b1c97cdcb64d26fd4948e98e1f8a99e2b40d25aec5f32R93-R107), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-e81088563f09768bf87b1c97cdcb64d26fd4948e98e1f8a99e2b40d25aec5f32L26-R33), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL139-R165), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL113-R108), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL84-R86), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL72-R76), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL38-R69), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-94767855b0f2244d37bdaef0e664cdfe27bd1263b627a42229aa9de3b178a22fL3-R55))
* Create `TeamAvatars` component to render a group of avatars for a team and add tests ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-b7376c30156f44e75e93b17b44cc6d8698a2e48bd3cd30d0b9adeb0e3734ced0R1-R66), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-b08f8ac3471a14e99426b520e2f33cfc712a0c707ef10ea015f2a5bd09e86098R1), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-b7376c30156f44e75e93b17b44cc6d8698a2e48bd3cd30d0b9adeb0e3734ced0R1-R66))
* Use `TeamAvatars` component in `TeamMenu` and `TeamSelect` components and update tests ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-6cfbc9590fb19d6c21035a68f21c238c5d1eef3c8d0ba9dd0f321d14dd5fa43dR56-R65), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-6cfbc9590fb19d6c21035a68f21c238c5d1eef3c8d0ba9dd0f321d14dd5fa43dR14-R16), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-6cfbc9590fb19d6c21035a68f21c238c5d1eef3c8d0ba9dd0f321d14dd5fa43dL2-R2), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d590d146adecfb00651739a5c231d8eca4ffcbf3fb7518782e1a565e2c462feaL96-R109), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d590d146adecfb00651739a5c231d8eca4ffcbf3fb7518782e1a565e2c462feaR17), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-dc36cd749cf11ee38972c057ed6d0c61c2de9f905ec33c3a017fcb496efaf0d3L87-R125), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-dc36cd749cf11ee38972c057ed6d0c61c2de9f905ec33c3a017fcb496efaf0d3L35-R72))
* Add `renderValue` prop to `Select` component in `TeamSelect` component to display team name or default value ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d590d146adecfb00651739a5c231d8eca4ffcbf3fb7518782e1a565e2c462feaR73))
* Add `sx` prop to `Avatar` component to accept custom style properties and update import statements ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d6514ce9ef6ce86c78cf729c141159454a64308888c0cd25fd7d15694627af79R41), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d6514ce9ef6ce86c78cf729c141159454a64308888c0cd25fd7d15694627af79L11-R18), [link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d6514ce9ef6ce86c78cf729c141159454a64308888c0cd25fd7d15694627af79R6))
* Add console log statement to `TeamSelect` component for debugging purposes ([link](https://github.com/JesusFilm/core/pull/1789/files?diff=unified&w=0#diff-d590d146adecfb00651739a5c231d8eca4ffcbf3fb7518782e1a565e2c462feaR40-R41))

### CODE REVIEW

File: apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx [hunk 1]
```diff
@@ -60,17 +60,24 @@ describe('TeamCreateDialog', () => {
...
-          lastActiveTeamId: null
+          lastActiveTeamId: 'teamId'
         }
       }
     }
   }
   function TestComponent(): ReactElement {
     const { activeTeam } = useTeam()
-
+    console.log(activeTeam?.title)
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx:80]: drop

File: apps/journeys-admin/src/components/Team/TeamCreateForm/TeamCreateForm.spec.tsx [hunk 2]
```diff
@@ -115,7 +122,7 @@ describe('TeamCreateForm', () => {
       )
     )
     expect(cache.extract()?.ROOT_QUERY?.teams).toEqual([
-      { __ref: 'Team:teamId1' },
+      { __ref: 'Team:teamId' },
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamCreateForm/TeamCreateForm.spec.tsx:125]: why did this get changed?

File: apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx [hunk 3]
```diff
@@ -105,7 +112,7 @@ describe('TeamCreateDialog', () => {
     )
     await waitFor(() => expect(handleClose).toHaveBeenCalled())
     expect(cache.extract()?.ROOT_QUERY?.teams).toEqual([
-      { __ref: 'Team:teamId1' },
+      { __ref: 'Team:teamId' },
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx:115]: why did this get changed

File: apps/journeys-admin/src/components/Team/TeamProvider/TeamProvider.tsx [hunk 4]
```diff
@@ -57,7 +67,7 @@ export function TeamProvider({ children }: TeamProviderProps): ReactElement {
     }
   )
 
-  function setActiveTeam(team: Team | null): void {
+  function setActiveTeam(team: Team | null | TeamCreate): void {
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamProvider/TeamProvider.tsx:70]: why does teamCreate need to be added in here? They should be compatible.

File: apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx [hunk 5]
```diff
@@ -94,12 +96,17 @@ export function TeamSelect({ onboarding }: TeamSelectProps): ReactElement {
...
+                  display: 'flex',
+                  flexDirection: 'row',
+                  alignItems: 'flex-start',
+                  justifyContent: 'space-between',
                   whiteSpace: 'normal',
                   wordWrap: 'break-word'
                 }}
               >
-                {team.title}
+                <Typography sx={{ maxWidth: '77%' }}>{team.title}</Typography>
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx:108]: this should just be flex.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Team/TeamCreateForm/TeamCreateForm.spec.tsx (hunk 2 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/Team/TeamCreateForm/TeamCreateForm.spec.tsx:125]: gql query change breaks this test

File: apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx (hunk 3 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/Team/TeamCreateDialog/TeamCreateDialog.spec.tsx:115]: gql query change breaks this test

File: apps/journeys-admin/src/components/Team/TeamProvider/TeamProvider.tsx (hunk 4 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/Team/TeamProvider/TeamProvider.tsx:70]: added this to resolve this type error: 

image

File: apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx (hunk 5 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx:108]: No word wrap as per design requirements without madwidth prop

File: apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx [hunk 6]
```diff
@@ -68,6 +69,7 @@ export function TeamSelect({ onboarding }: TeamSelectProps): ReactElement {
             displayEmpty
             value={activeTeam?.id ?? ''}
             onChange={handleChange}
+            renderValue={() => activeTeam?.title ?? 'Shared With Me'}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Team/TeamSelect/TeamSelect.tsx:72]: wrap in t('string')

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Review by tataihono — APPROVED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**